classes/Provider/Gateways/WebGatewayProvider.php

```diff
@@ -32,6 +49,134 @@ class WebGatewayProvider implements BootableProviderInterface, ServiceProviderIn
 {
     public function register(Container $app)
     {
+        // Controllers
+        $app[DashboardController::class] = function ($app) {
+            return new DashboardController($app['application.speakers'], $app['twig'], $app['url_generator']);
+        };
+        $app[ForgotController::class] = function ($app) {
+            return new ForgotController(
+                $app['form.factory'],
+                $app[AccountManagement::class],
+                $app['reset_emailer'],
+                $app['twig'],
+                $app['url_generator']
+            );
+        };
+        $app[PagesController::class] = function ($app) {
+            return new PagesController($app['twig'], $app['url_generator']);
+        };
+        $app[ProfileController::class] = function ($app) {
+            return new ProfileController(
+                $app[Authentication::class],
+                $app['purifier'],
+                $app['profile_image_processor'],
+                $app['twig'],
+                $app['url_generator']
+            );
+        };
+        $app[SecurityController::class] = function ($app) {
+            return new SecurityController(
+                $app[Authentication::class],
+                $app['twig'],
+                $app['url_generator']
+            );
+        };
+        $app[SignupController::class] = function ($app) {
+            return new SignupController(
+                $app[Authentication::class],
+                $app[AccountManagement::class],
+                $app[CallForPapers::class],
+                $app['twig'],
+                $app['url_generator']
+            );
+        };
+        $app[TalkFormController::class] = function ($app) {
+            return new TalkFormController(
+                $app[Authentication::class],
+                $app[TalkHelper::class],
+                $app[CallForPapers::class],
+                $app['twig'],
+                $app['url_generator']
+            );
+        };
+        $app[TalkProcessingController::class] = function ($app) {
+            return new TalkProcessingController(
+                $app[Authentication::class],
+                $app[TalkHelper::class],
+                $app[CallForPapers::class],
+                $app['purifier'],
+                $app['mailer'],
+                $app['twig'],
+                $app['url_generator'],
+                $app->config('application.email'),
+                $app->config('application.title'),
```

Comment:
I'm wondering if we can extract config to its own class, and inject that, rather than doing it this way.

Probably outside of the scope for this PR though.

Reply:
@BackEndTea 

We need to be careful that we don't turn the configuration into just another container - maybe there's an opportunity to create a bunch of value objects instead. 👍

Reply:
Yes, we could do that. However, in Symfony I'd convert these into parameters that would be injected via the DIC. I'm not a big friend on configuration objects, tbh.

Reply:
@derrabus So how would that work? Please keep in mind I am not an experienced Symfony user, having only ever built this thing using Silex and never a more robust Symfony app.

Reply:
@BackEndTea @localheinz I think decoupling the config object from being a property of an application object would be a good thing. Value objects containing these configuration values sound like a good idea to me.

Reply:
@chartjes When building the container, you would parse your configuration file and add parameters to the container builder.
```php
$container->setParameter('application.title', $config['application']['title'];
```
This allows you to refer to that parameter in your service definition:
```yaml
services:
    OpenCFP\Http\Controller\TalkProcessingController:
    public: true
    autowire: true
    bind:
        $applicationEmail: '%application.email%'
        $applicationTitle: '%application.title%'
        $applicationEndDate: '%application.enddate%'
```
This way, the configuration is directly compiled into the container, so you don't have to parse the configuration file on every request.

Reply:
> Yes, we could do that. However, in Symfony I'd convert these into parameters that would be injected via the DIC. I'm not a big friend on configuration objects, tbh.

This is exactly what I've started doing.